File: v131016/TeeNew/HighLow_Series.pas
unit HighLow_Series;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeCanvas, TeEngine, ErrorBar, TeeProcs, Chart, TeePenDlg;

type
  THighLowForm = class(TBaseForm)
    Series1: THighLowSeries;
    ButtonPen1: TButtonPen;
    ButtonPen2: TButtonPen;
    ButtonPen3: TButtonPen;
    Button1: TButton;
    Button2: TButton;
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeBrushDlg;

procedure THighLowForm.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(20);
  ButtonPen1.LinkPen(Series1.HighPen);
  ButtonPen2.LinkPen(Series1.LowPen);
  ButtonPen3.LinkPen(Series1.Pen);
end;

procedure THighLowForm.Button1Click(Sender: TObject);
begin
  EditChartBrush(Self,Series1.HighBrush);
end;

procedure THighLowForm.Button2Click(Sender: TObject);
begin
  EditChartBrush(Self,Series1.LowBrush);
end;

initialization
  RegisterClass(THighLowForm);
end.

File: TeeNew/Builder2007/HighLow_Series.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "HighLow_Series.h"
#include <TeeBrushDlg.hpp>
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "ErrorBar"
#pragma link "TeePenDlg"
#pragma resource "*.dfm"
THighLowForm *HighLowForm;
//---------------------------------------------------------------------------
__fastcall THighLowForm::THighLowForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall THighLowForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(20);
  ButtonPen1->LinkPen(Series1->HighPen);
  ButtonPen2->LinkPen(Series1->LowPen);
  ButtonPen3->LinkPen(Series1->Pen);
}
//---------------------------------------------------------------------------

void __fastcall THighLowForm::Button1Click(TObject *Sender)
{
  EditChartBrush(this,Series1->HighBrush);
}
//---------------------------------------------------------------------------

void __fastcall THighLowForm::Button2Click(TObject *Sender)
{
  EditChartBrush(this,Series1->LowBrush);
}
//---------------------------------------------------------------------------

